lm_eval/models/huggingface.py

```diff
@@ -441,20 +443,15 @@ def tokenizer_name(self) -> str:
     def _get_backend(
         self,
         config: Union[transformers.PretrainedConfig, transformers.AutoConfig],
-        backend: Optional[Literal["default", "causal", "seq2seq"]] = "default",
+        backend: Literal["default", "causal", "seq2seq"] = "default",
         trust_remote_code: Optional[bool] = False,
     ) -> None:
         """
         Helper method during initialization.
-        Determines the backend ("causal" (decoder-only) or "seq2seq" (encoder-decoder))
-        model type to be used.
+        Determines the backend ("causal" (decoder-only) or "seq2seq" (encoder-decoder)) model type to be used.
         sets `self.AUTO_MODEL_CLASS` appropriately if not already set.
+        Should only be called if isinstance(pretrained, str)! otherwise pass `backend` appropriately
         """
-        # escape hatch: if we're using a subclass that shouldn't follow
-        # the default _get_backend logic,
-        # then skip over the method.
-        # TODO: this seems very much undesirable in some cases--our code in HFLM
-        # references AutoModelForCausalLM at times to check for equality
         if self.AUTO_MODEL_CLASS is not None:
             return
```

Comment:
As per above comments on change in interface--this exit-early from the `_get_backend()` fn means we will now force users who override `AUTO_MODEL_CLASS` to set `self.backend` themselves in the init.

I think instead we should drop

```
        if self.AUTO_MODEL_CLASS is not None:
            return
```
entirely, and let our code handle things **but we should only set AUTO_MODEL_CLASS in the `_get_backend()` function if AUTO_MODEL_CLASS is currently None** !

That way, for subclasses that set AUTO_MODEL_CLASS themselves, we can use our existing "use user-specified backend if provided, else choose causal or seq2seq if can detect it among HF models, else fall back to assume causal" logic here (which should be what we want, so long as we signpost this behavior with logger msgs enough) and yet preserve the desired usage of having a special `AUTO_MODEL_CLASS` that gets used for model init. Sound good?

(as is, this PR I think breaks HF multimodal LMs at the moment. with the changes in my comments I believe it will no longer?)